src/api/api.ts

```diff
@@ -2,7 +2,7 @@ import { AddExpenseData, AddMonthData, LoginData, SignupData } from "../componen
 import { calculateDateForEachMonth } from "../components/utils/utils";
 import { supabase } from "../supabaseClient";
 
-export const addUser = async (values:SignupData) => {
+export const addUser = async (values:SignupData, toggleAlertSuccess: { (alert: string): void; (arg0: string): void; }, toggleAlertError: { (alert: string): void; (arg0: string): void; }) => {
```

Comment:
These type definitions look quite complicated. I think it's better to create interfaces for them :)